### PR TITLE
Fogl 820_sending_process_signals_handling

### PR DESCRIPTION
**Pull request by sstefano69**

### CODE REVIEW

File: python/foglamp/tasks/north/sending_process.py [hunk 1]
```diff
@@ -683,22 +704,32 @@ def send_data(self, stream_id):
...
+
                 try:
                     data_sent = self._send_data_block(stream_id)
                 except Exception as e:
                     data_sent = False
                     _message = _MESSAGES_LIST["e000021"].format(e)
                     SendingProcess._logger.error(_message)
+
+                # Terminates the execution in case a signal has been received
+                if SendingProcess._stop_execution:
```
Comment (praveen-garg) [inline python/foglamp/tasks/north/sending_process.py]: I believe this should move to top in `while` before `data_sent = self._send_data_block(stream_id)`

File: python/foglamp/tasks/north/sending_process.py [hunk 2]
```diff
@@ -846,6 +877,14 @@ def start(self):
         global _log_performance
         global _LOGGER
 
+        # Setups signals handlers, to properly handle the termination
+        # a) SIGINT: Keyboard interrupt
+        # b) SIGTERM: kill or system shutdown
+        # c) SIGHUP: Controlling shell exiting
+        signal.signal(signal.SIGINT, SendingProcess._signal_handler)
+        signal.signal(signal.SIGTERM, SendingProcess._signal_handler)
+        signal.signal(signal.SIGHUP, SendingProcess._signal_handler)
```
Comment (praveen-garg) [inline python/foglamp/tasks/north/sending_process.py]: Okay, but I assume SIGTERM shall be enough (as we are removing ctrl+c binding)

File: python/foglamp/tasks/north/sending_process.py [hunk 3]
```diff
@@ -245,6 +246,9 @@ class SendingProcess:
 
     _logger = None  # type: logging.Logger
 
+    _stop_execution = False
+    """ sets to True when a signal is captured and a graceful termination is needed """
```
Comment (praveen-garg) [inline python/foglamp/tasks/north/sending_process.py]: can remove `graceful` :)

File: python/foglamp/tasks/north/sending_process.py [hunk 4]
```diff
@@ -683,22 +704,32 @@ def send_data(self, stream_id):
...
         Raises:
-        Todo:
         """
         SendingProcess._logger.debug("{0} - ".format("send_data"))
         try:
             start_time = time.time()
             elapsed_seconds = 0
+
             while elapsed_seconds < self._config['duration']:
+
```
Comment (praveen-garg) [inline python/foglamp/tasks/north/sending_process.py]: avoid extra lines, making file too long to read

File: python/foglamp/tasks/north/sending_process.py [hunk 5]
```diff
@@ -683,22 +704,32 @@ def send_data(self, stream_id):
...
                 try:
                     data_sent = self._send_data_block(stream_id)
                 except Exception as e:
                     data_sent = False
                     _message = _MESSAGES_LIST["e000021"].format(e)
                     SendingProcess._logger.error(_message)
+
+                # Terminates the execution in case a signal has been received
+                if SendingProcess._stop_execution:
+
```
Comment (praveen-garg) [inline python/foglamp/tasks/north/sending_process.py]: avoid extra newline

File: python/foglamp/tasks/north/sending_process.py [hunk 6]
```diff
@@ -355,6 +359,23 @@ def __init__(self):
...
+        Args:
+        Returns:
+        Raises:
+        """
+
+        SendingProcess._stop_execution = True
+
+        short_stack_frame = str(_stack_frame)[:100]
+        SendingProcess._logger.debug("{func} - signal captured |{signo}| - info |{ssf}| ".format(
```
Comment (praveen-garg) [inline python/foglamp/tasks/north/sending_process.py]: signo > signal_num?, can we avoid ssf,  purpose?

File: python/foglamp/tasks/north/sending_process.py (hunk 3 above)
Comment (sstefano69) [reply python/foglamp/tasks/north/sending_process.py]: done

File: python/foglamp/tasks/north/sending_process.py (hunk 6 above)
Comment (sstefano69) [reply python/foglamp/tasks/north/sending_process.py]: done

File: python/foglamp/tasks/north/sending_process.py (hunk 1 above)
Comment (sstefano69) [reply python/foglamp/tasks/north/sending_process.py]: done

File: python/foglamp/tasks/north/sending_process.py (hunk 5 above)
Comment (sstefano69) [reply python/foglamp/tasks/north/sending_process.py]: done

**Comment by praveen-garg** [comment]
@sstefano69 could you please rebase with develop.

**Pull request merged**